socket/socketServer.js

```diff
@@ -0,0 +1,46 @@
+const db = require('../models/index')
+const { User } = db
+
+module.exports = function (io) {
+  io.use(async (socket, next) => {
+    // authentication
+    const userId = socket.handshake.userId // frontend - socket.userId = 1; socket.connect();
```

Comment:
I prefer verifying with jwt, not sure passport can work fine, theoretically, it should.
ttps://stackoverflow.com/questions/36788831/authenticating-socket-io-connections-using-jwt
This one is using local strategy
https://socket.io/docs/v3/middlewares/